mesonbuild/compilers/clike.py

```diff
@@ -170,10 +171,26 @@ def _get_search_dirs(self, env):
             stdo = p.stdo
         return stdo
 
-    @staticmethod
-    def _split_fetch_real_dirs(pathstr, sep=':'):
-        paths = []
-        for p in pathstr.split(sep):
+    def _split_fetch_real_dirs(self, pathstr):
+        # We need to use the path separator used by the compiler for printing
+        # lists of paths ("gcc --print-search-dirs"). By default
+        # we assume it uses the platform native separator.
+        pathsep = os.pathsep
+        # cygwin gcc is special and uses unix paths and ':' on Windows
+        gcc_cygwin = getattr(self, 'compiler_type', None) == CompilerType.GCC_CYGWIN
+        if gcc_cygwin:
+            pathsep = ':'
```

Comment:
Unless there's something horribly, horribly wrong, os.pathsep should be ':'  under cygwin (since it's doing it's hardest to pretend to be unix...)

Reply:
Oh, I kinda assumed that meson with normal Python might also handle cygwin gcc, but now tha tI think about it...